src/components/NavBar/NavBar.tsx

```diff
@@ -8,6 +9,19 @@ type TNavBar = {
   }[];
 };
 
+function CustomLink<C extends React.ElementType>(
+  props: LinkProps<C, { component?: C }>
+) {
+  return (
+    <Link
+      {...props}
+      style={({ isActive }: { isActive: boolean }) =>
+        isActive ? { textDecoration: 'underline' } : undefined
+      }
+    />
+  );
+}
+
```

Comment:
// I created a generic custom component so that I can use the NavLink "style" prop to display a specific style when the link is active or not. I followed the [MUI docs here](https://mui.com/material-ui/guides/composition/#with-typescript) to do this.
Another solution would have been to utilize `react-router-dom`'s `useLocation` to determine the `location.pathname`, and then change the underline style when mapping through each link further below if the `location.pathname === href`.